src/Bounds.cpp

```diff
@@ -733,6 +733,11 @@ class Bounds : public IRVisitor {
         }
     }
 
+    void visit(const Shuffle *op) {
+        // TODO: We could consider the bounds of the lanes used by the Shuffle.
```

Comment:
So you would iterate over those three things. If they happen to be ramps or broadcasts or Calls to a Func then something then Bounds could do something useful. Nothing would ever iterate over 384 things.